File: agentpit/datastructures/event.py
from typing import Optional
from pydantic import BaseModel

from agentpit.common import check_state


class Event(BaseModel):
    event_id: int
    slug: str
    title: str
    description: str = ""
    icon_url: Optional[str] = None
    category: Optional[str] = None
    start_date: Optional[int] = None
    end_date: Optional[int] = None
    polymarket_event_id: Optional[str] = None
    # Upstream Polymarket 24h volume (event- or series-level), captured at sync
    # time. Drives the homepage ordering. None when never synced from upstream.
    volume_24hr: Optional[float] = None
    # Upstream all-time volume, captured at the same time. This is the figure
    # the cards display; volume_24hr stays the ranking key.
    volume: Optional[float] = None

    def model_post_init(self, _context):
        check_state(len(self.title) > 0, "Event title must not be empty")
        check_state(len(self.slug) > 0, "Event slug must not be empty")

File: ui/src/types/event.ts
import type { Market } from "@/types/market";

export interface Event {
  event_id: number;
  slug: string;
  title: string;
  description: string;
  icon_url: string | null;
  category: string | null;
  start_date: number | null;
  end_date: number | null;
  polymarket_event_id: string | null;
  /** Upstream 24h volume in USD; null when never synced from upstream. */
  volume_24hr: number | null;
  /** Upstream all-time volume in USD; null when never synced from upstream.
   *  This is the figure the cards show — volume_24hr stays the ranking key. */
  volume: number | null;
  /** Order-book depth in USD; null when never synced from upstream. */
  liquidity: number | null;
  /** How contested the odds are, 0..1; null when never synced. */
  competitive: number | null;
}

export interface EventWithMarkets {
  event: Event;
  markets: Market[];
}

export interface ListEventsResponse {
  events: EventWithMarkets[];
  /** Server-side offset of the NEXT page, in wire units — client-side
   *  filtering (hidden resolved events) must not shift server paging. */
  nextOffset: number;
  /** True while the server keeps returning full pages. */
  hasMore: boolean;
  limit: number;
  offset: number;
}

export interface ListEventCategoriesResponse {
  categories: string[];
}
